File: spec/java/src/io/kaitai/struct/spec/TestYamlInts.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.YamlInts;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestYamlInts extends CommonSpec {
    @Test
    public void testYamlInts() throws Exception {
        YamlInts r = YamlInts.fromFile(SRC_DIR + "fixed_struct.bin");

        assertEquals(r.testU4Dec().intValue(), 0xffffffff);
        assertEquals(r.testU4Hex().intValue(), 0xffffffff);
        assertEquals(r.testU8Dec().longValue(), 0xffffffffffffffffL);
        assertEquals(r.testU8Hex().longValue(), 0xffffffffffffffffL);
    }
}

File: spec/cpp_stl_98/test_yaml_ints.cpp
#include <boost/test/unit_test.hpp>

#include <yaml_ints.h>

#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_yaml_ints) {
    std::ifstream ifs("src/fixed_struct.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    yaml_ints_t* r = new yaml_ints_t(&ks);

    BOOST_CHECK_EQUAL(r->test_u4_dec(), 0xffffffff);
    BOOST_CHECK_EQUAL(r->test_u4_hex(), 0xffffffff);
    BOOST_CHECK_EQUAL(r->test_u8_dec(), 0xffffffffffffffffUL);
    BOOST_CHECK_EQUAL(r->test_u8_hex(), 0xffffffffffffffffUL);

    delete r;
}
